saleor/webhook/transport/asynchronous/transport.py

```diff
@@ -624,6 +629,85 @@ def send_webhook_request_async(self, event_delivery_id) -> None:
     clear_successful_delivery(delivery)
 
 
+@app.task(
+    queue=settings.WEBHOOK_CELERY_QUEUE_NAME,
+    bind=True,
+)
+@allow_writer()
+def send_webhooks_async_for_app(
+    self,
+    app_id,
+) -> None:
+    domain = get_domain()
+    deliveries = get_deliveries_for_app(app_id)
+
+    if not deliveries:
+        return
+
+    for delivery_id, delivery_with_count in deliveries.items():
+        delivery = delivery_with_count.delivery
+        attempt_count = delivery_with_count.count
+
+        webhook = delivery.webhook
+        attempt = create_attempt(delivery, self.request.id)
+
+        try:
+            if not delivery.payload:
+                raise ValueError(f"Event delivery id: {delivery_id} has no payload.")
+            data = delivery.payload.get_payload()
+            # Convert payload to bytes if it's not already.
+            data = data if isinstance(data, bytes) else data.encode("utf-8")
+            # Count payload size in bytes.
+            payload_size = len(data)
+
+            with webhooks_opentracing_trace(
+                delivery.event_type,
+                domain,
+                payload_size,
+                app=webhook.app,
+            ):
+                response = send_webhook_using_scheme_method(
+                    webhook.target_url,
+                    domain,
+                    webhook.secret_key,
+                    delivery.event_type,
+                    data,
+                    webhook.custom_headers,
+                )
+
+            if response.status == EventDeliveryStatus.FAILED:
+                attempt_update(attempt, response)
+                if attempt_count >= 5:
+                    delivery_update(delivery, EventDeliveryStatus.FAILED)
+            elif response.status == EventDeliveryStatus.SUCCESS:
+                task_logger.info(
+                    "[Webhook ID:%r] Payload sent to %r for event %r. Delivery id: %r",
+                    webhook.id,
+                    sanitize_url_for_logging(webhook.target_url),
+                    delivery.event_type,
+                    delivery.id,
+                )
+                delivery.status = EventDeliveryStatus.SUCCESS
+                # update attempt without save to provide proper data in observability
+                attempt_update(attempt, response, with_save=False)
+        except ValueError as e:
+            response = WebhookResponse(
+                content=str(e), status=EventDeliveryStatus.FAILED
+            )
+            attempt_update(attempt, response)
+            if attempt_count >= 5:
+                delivery_update(delivery, EventDeliveryStatus.FAILED)
+
+        observability.report_event_delivery_attempt(attempt)
+        clear_successful_delivery(delivery)
+
+    send_webhooks_async_for_app.apply_async(
+        kwargs={
+            "app_id": app_id,
+        },
+    )
```

Comment:
We need some condition before we re-trigger. Please check if there are any pending deliveries left. otherwise it will run in infinite loop